src/cljs/nr/gamelobby.cljs

```diff
@@ -33,22 +34,45 @@
               (:date game)])
            games))
 
+
+(defn process-games-update 
+  [{:keys [diff notification] :as msg}]
+  (swap! app-state update-in [:games]
+          (fn [games]
+            (let [gamemap (into {} (map #(assoc {} (:gameid %) %) games))
+                  update-diff (reduce-kv
+                                (fn [m k v]
+                                  (assoc m k (merge (get m k {}) v)))
+                                gamemap
+                                (:update diff))
+                  delete-diff (apply dissoc update-diff (:delete diff))]
+              (sort-games-list (vals delete-diff)))))
+  (when (and notification (not (:gameid @app-state)))
+    (play-sound notification)))
+
 (ws/register-ws-handler!
   :games/list
-  #(swap! app-state assoc :games (sort-games-list %)))
+  (fn [msg]
+    (let [gamemap (into {} (map #(assoc {} (:gameid %) %) msg))]
+      (process-games-update {:diff {:update gamemap
+                                    :delete (map #(:gameid %) (filter #(not (get gamemap (:gameid %) false)) (:games @app-state)))}}))))
```

Comment:
this is an excellent place for 1) putting the bulk of this into a temp variable in the let block above, and 2) using a `->>` macro